rpc/core/tx.go

```diff
@@ -101,5 +103,29 @@ func TxSearch(ctx *rpctypes.Context, query string, prove bool, page, perPage int
 		}
 	}
 
+	if len(apiResults) > 1 {
+		for _, item := range apiResults {
```

Comment:
can you explain what this loop does? it looks like it just appends apiResults to itself.

Reply:
Yes, you are correct @melekes. But, it will remove the nil items present in the array. So, we will only get the txs which are not null. If we will allow nil items then, at the time of order by it will not able to sort correctly and it will fail because, the one item from (a,b) is nil.